common/inbus/inbus.py

```diff
@@ -21,42 +21,50 @@ def person_by_login(login: str) -> dto.PersonSimple | None:
 
     return person_simple
 
 
-def subject_versions(department_id: dto.DepartmentId = 386) -> List[dto.SubjectVersion]:
+def subject_versions(department_id: dto.DepartmentId = dto.DepartmentId(386)) -> List[dto.SubjectVersion]:
     """
     Get list of all subjects and their versions by department.
     Here `386` is Department of Computer Science.
     """
-    results_per_page = 20
+    results_per_page: int = 20
     subject_versions = []
-    offset = 0
+    offset: int = 0
     while True:
         url = urllib.parse.urljoin(config.INBUS_SERVICE_EDISON_URL, 'edu/subjectVersions')
         subject_versions_resp = utils.inbus_request(url, {'departmentId': department_id, 'offset': offset, 'limit': results_per_page})
 
-        subject_versions_json = subject_versions_resp.json()
+        if subject_versions_resp:
+            subject_versions_json = subject_versions_resp.json()
 
-        for subject_version_json in subject_versions_json:
-            subject_json = subject_version_json['subject']
-            subject_guarantee = serde.from_dict(dto.Person, subject_json['guarantee'])
-            subject_version_guarantee = serde.from_dict(dto.Person, subject_version_json['guarantee'])
-            subject = dto.Subject(subjectId=subject_json['subjectId'], code=subject_json['code'], abbrev=subject_json['abbrev'],
-                                  title=subject_json['title'], guarantee=subject_guarantee)
-            subject_version = dto.SubjectVersion(subjectVersionId=subject_version_json['subjectVersionId'], subject=subject,
-                                                 subjectVersionCompleteCode=subject_version_json['subjectVersionCompleteCode'], guarantee=subject_version_guarantee)
+            for subject_version_json in subject_versions_json:
+                subject_json = subject_version_json['subject']
+                subject_guarantee = serde.from_dict(dto.Person, subject_json['guarantee'])
+                subject_version_guarantee = serde.from_dict(dto.Person, subject_version_json['guarantee'])
+                subject = dto.Subject(subjectId=subject_json['subjectId'], code=subject_json['code'],
+                                      abbrev=subject_json['abbrev'], title=subject_json['title'],
+                                      guarantee=subject_guarantee)
+                subject_version = dto.SubjectVersion(subjectVersionId=subject_version_json['subjectVersionId'],
+                                                     subject=subject,
+                                                     subjectVersionCompleteCode=subject_version_json['subjectVersionCompleteCode'],
+                                                     guarantee=subject_version_guarantee)
 
-            subject_versions.append(subject_version)
+                subject_versions.append(subject_version)
 
-        results = len(subject_versions_json)
+            results: int = len(subject_versions_json)
```

Comment:
```suggestion
            results = len(subject_versions_json)
```
It's a matter of taste, but I find it to be a bit distracting to add typing annotations for obvious types (e.g. return value of `len`) for local variables. In a (reasonable) statically typed language, these annotations would be inferred anyway.

Reply:
Yeah, I got carried away by adding typing annotations :-)